structuralcodes/codes/ec2_2023/_section_8_1_bending.py

```diff
@@ -0,0 +1,593 @@
+        )
+    if f_cd < 0:
+        raise ValueError(
+            f'f_cd must be non-negative. Got {sigma_c2d} instead.'
+        )
+    if ddg < 0:
+        raise ValueError(f'ddg must be non-negative. Got {sigma_c2d} instead.')
+
+    if sigma_c2d <= 0.6 * f_cd:
+        delta_f_cd = 4 * sigma_c2d
+    else:
+        delta_f_cd = 3.5 * sigma_c2d ** (3 / 4) * f_cd ** (1 / 4)
+
+    if ddg < 32:
+        delta_f_cd *= ddg / 32
+
+    return delta_f_cd
+
+
+def confinement_sigma_c2d_circular_square(
+    A_s_conf: float, f_yd: float, b_cs: float, s: float
+) -> float:
+    """Calculate the confinement stress (sigma_c2d) for circular and square
+    members with single confinement reinforcement.
+
+    EN1992-1-1:2023 Eq (8.11).
+
+    Args:
+        A_s_conf (float): Cross-sectional area of one leg of confinement
+            reinforcement in mm2.
+        f_yd (float): Yield strength of reinforcement in MPa.
+        b_cs (float): Width of the confinement core in mm.
+        s (float): Spacing of confinement reinforcement in mm.
+
+    Returns:
+        float: Confinement stress in MPa.
+    """
+    if A_s_conf < 0:
+        raise ValueError(
+            f'A_s_conf must be non-negative. Got {A_s_conf} instead.'
+        )
+    if f_yd < 0:
+        raise ValueError(f'f_yd must be non-negative. Got {f_yd} instead.')
+    if b_cs < 0:
+        raise ValueError(f'b_cs must be non-negative. Got {b_cs} instead.')
+    if s < 0:
+        raise ValueError(f's must be non-negative. Got {s} instead.')
+
+    return 2 * A_s_conf * f_yd / (b_cs * s)
+
+
+def confinement_sigma_c2d_rectangular(
+    A_s_conf: float, f_yd: float, b_csx: float, b_csy: float, s: float
+) -> float:
+    """Calculate the confinement stress (sigma_c2d) for rectangular members
+    with single confinement reinforcement.
+
+    EN1992-1-1:2023 Eq. (8.12).
+
+    Args:
+        A_s_conf (float): Cross-sectional area of one leg of confinement
+            reinforcement.
+        f_yd (float): Yield strength of reinforcement.
+        b_csx (float): Width of the confinement core in x direction.
+        b_csy (float): Width of the confinement core in y direction.
+        s (float): Spacing of confinement reinforcement.
+
+    Returns:
+        float: Confinement stress.
+    """
+    if A_s_conf < 0:
+        raise ValueError(
+            f'A_s_conf must be non-negative. Got {A_s_conf} instead.'
+        )
+    if f_yd < 0:
+        raise ValueError(f'f_yd must be non-negative. Got {f_yd} instead.')
+    if b_csx < 0:
+        raise ValueError(f'b_csx must be non-negative. Got {b_csx} instead.')
+    if b_csy < 0:
+        raise ValueError(f'b_csy must be non-negative. Got {b_csy} instead.')
+    if s < 0:
+        raise ValueError(f's must be non-negative. Got {s} instead.')
+
+    return 2 * A_s_conf * f_yd / (max(b_csx, b_csy) * s)
+
+
+def confinement_sigma_c2d_multiple(
+    A_s_confx: List[float],
+    A_s_confy: List[float],
```

Comment:
Is it better to refer to ArrayLike? And then deal a single value or tuple/list or anything transformable to array? In this way the user could provide the list, or directly the summation?